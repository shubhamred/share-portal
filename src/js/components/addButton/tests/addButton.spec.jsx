import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AddButton from '../addButton';

configure({ adapter: new Adapter() });

describe('components/Add Button', () => {
  it('Renders default options', () => {
    const component = shallow(
      <AddButton />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('handles click', () => {
    const props = {
      onClick: jest.fn()
    };
    const component = shallow(
      <AddButton onClick={props.onClick} />
    );
    component.simulate('click');
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
