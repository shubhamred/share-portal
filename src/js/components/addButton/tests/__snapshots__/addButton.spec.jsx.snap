// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Add Button Renders default options 1`] = `
<WithStyles(ForwardRef(Grid))
  onClick={[Function]}
  props={
    Object {
      "label": "Add",
      "onClick": [Function],
    }
  }
  style={
    Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Memo(ForwardRef(AddCircleIcon))
    style={
      Object {
        "fill": "#5064E2",
      }
    }
  />
  <WithStyles(ForwardRef(Grid))
    style={
      Object {
        "color": "#164363",
        "padding": "4px",
      }
    }
  >
    Add
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
